CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+project(bladebit)
+
+set(CMAKE_CONFIGURATION_TYPES release debug)
+
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "release"
+        CACHE STRING "Possible values are: release, debug"
+        FORCE
+   )
+endif()
+message("Config: ${CMAKE_BUILD_TYPE}")
+
+
+# Grab BLS
+include(FetchContent)
+
+set(BUILD_BLS_PYTHON_BINDINGS "0" CACHE STRING "0")
+set(BUILD_BLS_TESTS "0" CACHE STRING "")
+set(BUILD_BLS_BENCHMARKS "0" CACHE STRING "")
+
+FetchContent_Declare(
+  bls
+  GIT_REPOSITORY https://github.com/Chia-Network/bls-signatures.git
+  GIT_TAG        1.0.6
+)
+FetchContent_MakeAvailable(bls)
+
+set(CMAKE_C_FLAGS_RELEASE_INIT)
+set(CMAKE_C_FLAGS_DEBUG_INIT)
+set(CMAKE_CXX_FLAGS_RELEASE_INIT)
+set(CMAKE_CXX_FLAGS_DEBUG_INIT)
```

Comment:
Setting CMAKE_*_INIT to empty doesn't have that effect across all setups, so whatever CMake puts in them stays. I would also consider just extending the initial flags with what we need?